modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java

```diff
@@ -298,6 +299,9 @@ private MuleEvent innerProcess(MuleEvent muleEvent, boolean checkRetry) throws M
         {
             validateResponse(muleEvent);
         }
+        // Set RequestContext ThreadLocal in new thread for backwards compatibility
+        OptimizedRequestContext.unsafeSetEvent(muleEvent);
```

Comment:
The issue is caused by the HttpListener not setting the event in the RequestContext.  If the event was already set there would be nothing to do here, because i) the HttpRequester mutates the event but doesn't create a new instance ii) there isn't a thread boundary